src/operator/controllers/external_traffic/network_policy_uploader.go

```diff
@@ -80,27 +77,32 @@ func (r *NetworkPolicyUploaderReconciler) Reconcile(ctx context.Context, req ctr
 		ctx, &podList,
 		&client.MatchingLabelsSelector{Selector: selector},
 		&client.ListOptions{Namespace: netpol.Namespace})
-
 	if err != nil {
 		logrus.WithError(err).Errorf("error when reading podlist")
 		return ctrl.Result{}, nil
 	}
 
+	if len(podList.Items) == 0 {
```

Comment:
I think this suggests that we need to also run this logic on the PodReconciler, as network policies may be present without any pods, and the pods may appear later.

Reply:
Agreed, from a short search it seem there are other location in the code who need it, so I suggest we make it a separate task